docker/gcp_pudl_etl.sh

```diff
@@ -12,128 +12,141 @@ function send_slack_msg() {
+        echo "Copying outputs to GCP distribution bucket"
+        gsutil -m -u "$GCP_BILLING_PROJECT" cp -r "$PUDL_OUTPUT/*" "gs://pudl.catalyst.coop/$GITHUB_REF"
+
+        echo "Removing old $GITHUB_REF outputs from AWS distributon bucket."
+        aws s3 rm "s3://pudl.catalyst.coop/$GITHUB_REF" --recursive
+        echo "Copying outputs to AWS distribution bucket"
+        aws s3 cp "$PUDL_OUTPUT/" "s3://pudl.catalyst.coop/$GITHUB_REF" --recursive
+    fi
 }
 
 function zenodo_data_release() {
     echo "Creating a new PUDL data release on Zenodo."
-    ~/devtools/zenodo/zenodo_data_release.py --publish --env sandbox --source-dir $PUDL_OUTPUT
+    ~/devtools/zenodo/zenodo_data_release.py --publish --env sandbox --source-dir "$PUDL_OUTPUT"
 }
 
-
 function notify_slack() {
     # Notify pudl-builds slack channel of deployment status
-    if [ $1 = "success" ]; then
+    if [ "$1" = "success" ]; then
         message=":large_green_circle: :sunglasses: :unicorn_face: :rainbow: The deployment succeeded!! :partygritty: :database_parrot: :blob-dance: :large_green_circle:\n\n "
         message+="<https://github.com/catalyst-cooperative/pudl/compare/main...${GITHUB_REF}|Make a PR for \`${GITHUB_REF}\` into \`main\`!>\n\n"
-    elif [ $1 = "failure" ]; then
+    elif [ "$1" = "failure" ]; then
         message=":large_red_square: Oh bummer the deployment failed ::fiiiiine: :sob: :cry_spin:\n\n "
     else
         echo "Invalid deployment status"
         exit 1
     fi
-    message+="See https://console.cloud.google.com/storage/browser/nightly-build-outputs.catalyst.coop/$ACTION_SHA-$GITHUB_REF for logs and outputs."
+    message+="See https://console.cloud.google.com/storage/browser/nightly-build-outputs.catalyst.coop/$BUILD_ID for logs and outputs."
 
     send_slack_msg "$message"
 }
 
 # # Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
 # 2>&1 redirects stderr to stdout.
-run_pudl_etl 2>&1 | tee $LOGFILE
+run_pudl_etl 2>&1 | tee "$LOGFILE"
 
 ETL_SUCCESS=${PIPESTATUS[0]}
 
 copy_outputs_to_gcs
 
 # if pipeline is successful, distribute + publish datasette
 if [[ $ETL_SUCCESS == 0 ]]; then
+    if [ "$GITHUB_ACTION_TRIGGER" = "schedule" ]; then
+        # Remove read-only authentication header added by git checkout
+        git config --unset http.https://github.com/.extraheader
+        git config user.email "pudl@catalyst.coop"
+        git config user.name "pudlbot"
+        git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
+        # Update the nightly branch to point at newly successful nightly build tag
+        git checkout nightly
+        git merge --ff-only "$NIGHTLY_TAG"
+        git push
+    fi
     # Deploy the updated data to datasette
-    if [ $GITHUB_REF = "dev" ]; then
-        python ~/devtools/datasette/publish.py 2>&1 | tee -a $LOGFILE
+    if [ "$GITHUB_REF" = "dev" ]; then
```

Comment:
Should this be rolled in to the above `if` block? We're using `"$GITHUB_REF" = "dev"` to mean "is scheduled nightly build", right?

Reply:
We could do that, but that would mean that the *only* way to redeploy Datasette is for the scheduled build to run. Whereas with this condition, if you deploy `dev` with `workflow_dispatch` it will redeploy Datasette upon success (and it will also be redeployed if the nightly builds succeed, since they only run on `dev`)

Reply:
But in general there's a lot of fuzziness with many of these conditionals, and it's not always clear what the intended behavior is, with some of them looking at `GITHUB_REF` and others looking at the action trigger.

Reply:
OK. We can leave this as is for now, but as we are consolidating onto `main` we'll have to change it.

I liked @rousik 's suggestion (somewhere...) of having a script with a lot of actual logical arguments, something sort of like
```
./run_the_frickin_nightly_build.py --etl-fast --deploy-to-datasette --publish-to-gcp --publish-to-aws --update-branch nightly
```

And then building that command line in the GHA workflow. So then at least it's clear what logical thing we mean by `"$GITHUB_REF" = "dev"`

Reply:
There are a bunch of hard-coded references to `dev` that we'll need to update.  I agree that setting all of the options in one place would be nice. And getting all of this into Python would be nice (I think we could ditch installing the AWS CLI and Google Cloud SDK with a simple deploy-to-buckets script). Definitely having 3-4 places with logic (workflow, Dockerfile, shell script, pudl_etl python script) isn't the best!